### DISCUSSION TITLE
Help getting gh cli to pull machine types for a repo

### DISCUSSION THREAD

**Discussion by shellerci** [community/community#10387] [Codespaces] [answered]
I'm having trouble figuring out the proper way to invoke the github command line utility to retrieve a list of available machine types for a repository.  I can make the same call with the same values with curl and get a response no problem, but I can't seem to make the cli play nice.

With curl : 

```zsh
~|⇒ curl --location \
  --request GET "https://api.github.com/repos/${OWNER}/${REPO}/codespaces/machines?location=EastUs" \
  --header "authorization: token ${GH_TOKEN}"
{
  "machines": [
    {
      "name": "basicLinux32gb",
      "display_name": "2 cores, 4 GB RAM, 32 GB storage",
      "operating_system": "linux",
      "storage_in_bytes": 34359738368,
      "memory_in_bytes": 4294967296,
      "cpus": 2,
      "prebuild_availability": null
    }
  ],
  "total_count": 1
}
```

With gh cli
```zsh
~|⇒ gh api "repos/${OWNER}/${REPO}/codespaces/machines -f location=EastUs"
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
gh: Not Found (HTTP 404)
```

The issue seems to be with how I'm passing the location param, if I leave off the -f argument I get

```zsh
~|⇒ gh api "repos/${OWNER}/${REPO}/codespaces/machines"
{
  "message": "You must provide a 'location' to fetch available machines",
  "documentation_url": "https://docs.github.com/rest/reference/codespaces#list-available-machine-types-for-a-repository"
}
gh: You must provide a 'location' to fetch available machines (HTTP 400)
```

I've tried variations with quoting -f and -F arguments, and attempting to pass it as json in the request body all with no success. 

Does anyone know what I'm doing wrong here?

**Comment by asciimike** [comment, marked as answer]
The answer comes from the [CLI docs](https://cli.github.com/manual/gh_api):

> Pass one or more -f/--raw-field values in "key=value" format to add static string parameters to the request payload. To add non-string or otherwise dynamic values, see --field below. Note that adding request parameters will automatically switch the request method to POST. To send the parameters as a GET query string instead, use --method GET.

`gh api --method GET "repos/{owner}/{repo}/codespaces/machines" -f location=EastUs` (`-X` also works)
